tests/unit/disco/components/TestAddon.js

```diff
@@ -386,22 +388,59 @@ describe(__filename, () => {
     });
   });
 
-  describe('<Addon type="theme"/>', () => {
+  describe.each([
+    {
+      type: ADDON_TYPE_THEME,
+      previews: [],
+      previewURL: 'https://addons.cdn.mozilla.net/full/54321.png',
+    },
+    {
+      type: ADDON_TYPE_STATIC_THEME,
+      previews: [
+        {
+          ...fakePreview,
+          image_url: 'https://addons.cdn.mozilla.net/full/54321.png',
+        },
+      ],
+    },
+  ])(`Addon for %o`, async (testConfig) => {
```

Comment:
I like your use of `.each` but I would like to see a better description. Currently it is:

```
  ● /Users/williamdurand/projects/mozilla/addons-frontend/tests/unit/disco/components/TestAddon.js › Addon for { type: 'statictheme',
  previews:
   [ { id: 1,
       caption: 'Image 1',
       image_url: 'https://addons.cdn.mozilla.net/full/54321.png',
       thumbnail_url: 'https://addons.cdn.mozilla.net/thumb/1.png',
       image_size: [ 400, 200, [length]: 2 ],
       thumbnail_size: [ 200, 100, [length]: 2 ] },
     [length]: 1 ] } › renders the correct image for a theme
```

it is not really readable. I think here that would make more sense to have to set of tests because you are not exactly testing the same things it seems (otherwise `previews` would have the same content, no?)

Alternatively, you can use and configure `previews` in the `beforeEach` block:

```
describe.each([ADDON_TYPE_THEME, ADDON_TYPE_STATIC_THEME])(`Addon with type = %s`, (type) => {
});
```

Reply:
I saw some of the same repeating tests but I see what you're saying. Let me try this way you suggest with just type.

Reply:
let me know if how it have it now is ok. I extracted some of it..